File: js/commonChars.js
function commonChars(A) {
    let AMap = {};
    let result = [];
    let ASet = new Set()
    for (let i = 0; i < A[0].length; ++i) {
        ASet.add(A[0][i]);
    }

    for (ch of ASet) {
        AMap[ch] = count(A[0], ch);
    }

    for (word of A) {
        for (ch of ASet) {
            if (count(word, ch)) {
                AMap[ch] = Math.min(count(word, ch), AMap[ch]);
            } else {
                if (AMap.hasOwnProperty(ch)) {
                    delete AMap[ch];
                }
            }
        }
    }

    for (let ch in AMap) {
        for (let i = 0; i < AMap[ch]; ++i) {
            result.push(ch);
        }
    }

    return result;

}

function count(str, val) {
    let counter = 0;
    for (let i = 0; i < str.length; ++i) {
        if (str[i] == val) {
            counter++;
        }
    }
    return counter;
}

const result = commonChars(["bella","label","roller"]);
console.log('hi');

File: py/commonChars.py

import math

def commonChars(A):
    AMap = dict()
    #AMapF = dict()
    listResult = []
    
    for ch in set(A[0]): # this is a cool trick for 'each unique item'
        AMap[ch] = A[0].count(ch)
    
    # for all remaining words now.. run counts of each..and if 0, then pop..otherwise add min to results
    for i in range(1, len(A)):
        for ch in set(A[0]):
            if ch in AMap:
                if ch in A[i]:
                    localCount = A[i].count(ch)
                    #AMapF[ch] = ch * min(localCount, AMap[ch])
                    AMap[ch] = min(localCount, AMap[ch])
                else:
                    if ch in AMap:
                        AMap.pop(ch)

    for ch in AMap:
        for i in range(0, AMap[ch]):
            listResult.append(ch)
    
    return listResult
    


result = commonChars(["acabcddd","bcbdbcbd","baddbadb","cbdddcac","aacbcccd","ccccddda","cababaab","addcaccd"])
print('hi')